src/server/project.ts

```diff
@@ -760,6 +850,16 @@ namespace ts.server {
             return this.configFileName;
         }
 
+        getExternalFiles(): string[] {
+            const items: string[] = [];
+            for (const plugin of this.plugins) {
+                if (plugin.getExternalFiles) {
+                    items.push(...plugin.getExternalFiles(this));

```

Comment:
If I'm not mistaken, this calls plugin code which can throw - any errors should probably be handled and logged?